addons/mail/static/src/owl/store/messaging_actions.js

```diff
@@ -1754,15 +1754,15 @@ const actions = {
                 await dispatch('_loadNewMessagesOnThread', threadLocalId);
             }
             await dispatch('_fetchThreadAttachments', threadLocalId);
-            if (chatter.activityIds) {
-                if (chatter.activityIds.length > 0) {
-                    chatter.activityLocalIds = await dispatch('_fetchActivities',
-                        chatter.activityIds,
-                        chatter.localId
-                    );
-                }
-                delete chatter.activityIds;
+            if (chatter.activityIds && chatter.activityIds.length > 0) {
+                chatter.activityLocalIds = await dispatch('_fetchActivities',
+                    chatter.activityIds,
+                    chatter.localId
+                );
+            } else {
+                chatter.activityLocalIds = [];
             }
+            delete chatter.activityIds;
```

Comment:
I know it was the previous code just moved, but it feels strange to have that key being temporarily set and then deleted after processing.

Shouldn't the processing be done immediately (in the action that introduces `activityIds` if there is such a thing) so that it never has to be added or deleted in the state?

I'm asking naively without the full knowledge of the code leading there, so let me know what you think.

Reply:
I think this part would be modified when entity is there. The problem here is that we have a _compute method that is used by the create and the update and we wanted to avoid to pass arguments to this method so it just need the store to compute everything.